lib/csapi/keys.h

```diff
@@ -166,6 +166,11 @@ struct JsonObjectConverter<QueryKeysJob::DeviceInformation> {
         fillFromJson<DeviceKeys>(jo, result);
         fromJson(jo.value("unsigned"_ls), result.unsignedData);
     }
+    static void dumpTo(QJsonObject& jo, const QueryKeysJob::DeviceInformation& deviceInformation)
+    {
+        jo = toJson<DeviceKeys>(deviceInformation);
+        //addParam<>(jo, "unsigned"_ls, deviceInformation.unsignedData);
+    }
```

Comment:
What do i need to do instead? put it somewhere else?

Reply:
Basically yes. The trick is that you're not (should not be) confined by this `JsonObjectConverter<>` specialisation; you can (should be able to) simply provide an overload of `auto toJson(const QueryKeysJob::DeviceInformation&)` instead. Now, as I just found out, this apparently doesn't work because the fallback `toJson()` that calls `JsonConverter<>::dumpTo` is not sufficiently SFINAE-friendly. Let me see if I can quickly fix this.

Reply:
Actually wait a minute - do you even need to store `DeviceInformation` or would storing a map-to-map-to-`DeviceKeys` be enough. I see that you're diving into this multi-storey map when parsing the result of `QueryKeysJob` anyway. Let me give a couple of suggestions in `connection.cpp`, and you'll be fine.